bentoml/service_env.py

```diff
@@ -164,7 +187,27 @@ def save(self, path):
         requirements_txt_file = os.path.join(path, "requirements.txt")
 
         with open(requirements_txt_file, "wb") as f:
-            pip_content = "\n".join(self._pip_dependencies).encode("utf-8")
+            dependencies_map = {}
+            for dep in self._pip_dependencies:
+                name, version = parse_requirement_string(dep)
+                dependencies_map[name] = version
+            if self._auto_pip_dependencies:
+                if "__main__" in sys.modules:
+                    main_module = sys.modules["__main__"]
+                    if hasattr(main_module, "__file__"):
```

Comment:
most of the time this is being called from a jupyter notebook, where the main module does not have a __file__ attribute

Reply:
this is also why in our notebook examples, we recommend users to use `%writefile` magic keyword to write the BentoService class definition to a separate file.